File: app/src/main/java/com/example/guestlog_eldroid/Guest.kt
package com.example.guestlog_eldroid
import com.google.gson.annotations.SerializedName

data class Guest(
    val id: String? = null,
    @SerializedName("name") var name: String,
    @SerializedName("email") var email: String,
    @SerializedName("time_in") val timeIn: String? = null,
    @SerializedName("date") val date: String? = null,
    @SerializedName("phone_number") var phoneNumber: String? = null,
    @SerializedName("age") var age: String = null.toString()
)

File: GuestLogAPI/app/Models/Guest.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Concerns\HasUuids; // For UUID support 

class Guest extends Model
{
    use HasFactory, HasUuids;

    protected $keyType = 'string'; // Set the key type to string
    public $incrementing = false; // Disable auto-incrementing
    protected $fillable = [
        'name',
        'email',
        'time_in',
        'date',
        'phone_number',
        'age',
    ];
    
    // Mutator for converting camelCase to snake_case
    public function setAttribute($key, $value)
    {
        $key = \Illuminate\Support\Str::snake($key);
        parent::setAttribute($key, $value);
    }
    
}
